src/timeMachine/components/view_options/TimeAutoToggle.tsx

```diff
@@ -0,0 +1,38 @@
+import React, {FC, useState} from 'react'
+import {useSelector} from 'react-redux'
+import {SlideToggle, ComponentSize} from '@influxdata/clockface'
+
+import {getXColumnTimeDomainBounds} from 'src/timeMachine/selectors'
+
+interface Props {
+  onSetDomain: (xDomain: [number, number]) => void
+}
+
+const TimeDomainAutoToggle: FC<Props> = ({onSetDomain}) => {
+  const [isActive, setIsActive] = useState<boolean>(false)
+
+  const timeDomain = useSelector(getXColumnTimeDomainBounds)
+
+  const handleClick = (): void => {
+    if (!isActive) {
+      onSetDomain([timeDomain[0], timeDomain[1]])
+    } else {
+      onSetDomain(null)
+    }
+    setIsActive(!isActive)
+  }
+  return (
+    <div data-testid="time-domain-toggle">
+      <SlideToggle
+        style={{height: 'auto'}}
+        size={ComponentSize.ExtraSmall}
+        active={isActive}
+        onChange={handleClick}
```

Comment:
kind of oddly named function. 'onChange, execute handleClick'. `handleClick` seems more like a function on a component, something like a function that handles someone clicking on the component. think about it like this: what if we wanted to add super power user keyboard shortcuts and one of those shortcuts let them toggle this component. `handleClick` makes not much sense in that case.

what is the function doing? to me it looks like it's toggling the component being `active` or not. See if you can name it something more along those lines